playbooks/byo/openshift-cluster/upgrades/docker/docker_upgrade.yml

```diff
@@ -23,11 +23,23 @@
   serial: 1
   any_errors_fatal: true
   tasks:
-  - name: Prepare for Node draining
-    command: >
-      {{ openshift.common.client_binary }} adm manage-node {{ openshift.node.nodename }} --schedulable=false
+  - name: Load lib_openshift modules
+    include_role:
+      name: lib_openshift
+
+  - name: Mark node unschedulable
+    oadm_manage_node:
+      node: "{{ openshift.node.nodename | lower }}"
+      schedulable: False
     delegate_to: "{{ groups.oo_first_master.0 }}"
-    when: l_docker_upgrade is defined and l_docker_upgrade | bool and inventory_hostname in groups.oo_nodes_to_upgrade
+    retries: 10
+    delay: 5
```

Comment:
Let me give a bit of history.  This is coming from the oadm command.  We found that at times, we would issue the  "oadm manage-node <nodename> --schedudable=false" (done through the module), but the command would fail.

We were in a rush to get this code working as it is the upgrade code we use on our customers.  This is a bug in openshift oadm command that we were compensating for.

The correct path is to file a bug upstream to get it fixed.  In the meantime, we put in the retry in to actually get the work done.

I agree with @detiber in that the module could be made more resilient, but the question would be what is the right thing for the module to do?  Do we retry 3 times? 5times?  i don't know the answer when you are trying to compensate for a bug.